src/sentry/static/sentry/app/components/forms/form.jsx

```diff
@@ -1,36 +1,97 @@
 import React from 'react';
+import underscore from 'underscore';
 
 import FormState from './state';
 import {t} from '../../locale';
 
-class Form extends React.Component {
+export default class Form extends React.Component {
+  static propTypes = {
+    onSubmit: React.PropTypes.func.isRequired,
+    onSubmitSuccess: React.PropTypes.func,
+    onSubmitError: React.PropTypes.func,
+    submitDisabled: React.PropTypes.bool,
+    submitLabel: React.PropTypes.string,
+    footerClass: React.PropTypes.string,
+    extraButton: React.PropTypes.element,
+    initialData: React.PropTypes.object
+  };
+
+  static defaultProps = {
+    submitLabel: t('Save Changes'),
+    submitDisabled: false,
+    footerClass: 'form-actions align-right',
+    className: 'form-stacked'
+  };
+
+  static childContextTypes = {
+    form: React.PropTypes.object
+  };
+
   constructor(props) {
     super(props);
     this.state = {
-      formData: Object.assign({}, this.props.initialData),
-      errors: {}
+      data: {...this.props.initialData},
+      errors: {},
+      initialData: {...this.props.initialData},
+      state: FormState.READY
     };
-    ['onSubmit', 'onFieldChange'].forEach(f => {
+    ['onSubmit', 'onSubmitSuccess', 'onSubmitError', 'onFieldChange'].forEach(f => {
```

Comment:
Another option, since you've added class properties transform to babel, is to   define them as a class prop and use arrow function i.e.

```
onSubmit = () => {
}
```